File: min_start_value.py
def min_start_value(nums: list[int]) -> int:
    min_value, curr = 0, 0

    for num in nums:
        curr += num
        min_value = min(min_value, curr)
    
    return 1 - min_value

print(min_start_value([-3,2,-3,4,2]))

File: minStartValue.js
function minStartValue(nums) {
    let minValue = 0;
    let curr = 0;

    for (let num of nums) {
        curr += num;
        minValue = Math.min(curr, minValue);
    }

    return 1 - minValue;
}

console.log(minStartValue([-3,2,-3,4,2]))